swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/events/ResourceStateManager.kt

```diff
@@ -162,37 +163,68 @@ class ResourceStateManager(
     if (event is OptOutResourceEvent) {
       log.debug("Tagging resource ${event.markedResource.uniqueId()} with \"expiration_time\":\"never\"")
       val taskId = tagResource(event.markedResource, event.workConfiguration)
-      log.debug("Tagging resource ${event.markedResource.uniqueId()} in {}", StructuredArguments.kv("taskId", taskId))
+      if (taskId != null) {
+        log.debug("Tagging resource ${event.markedResource.uniqueId()} in {}", StructuredArguments.kv("taskId", taskId))
+      }
     }
   }
 
   // todo eb: pull to another kind of ResourceTagger?
   private fun tagResource(
     resource: MarkedResource,
     workConfiguration: WorkConfiguration
-  ): String {
-    val taskId = taggingService.upsertImageTag(
-      UpsertImageTagsRequest(
-        imageNames = setOf(resource.name ?: resource.resourceId),
+  ): String? {
+    var taskId: String? = with(resource.resourceType) {
+      when {
+        equals("serverGroup", true) -> tagAsg(resource, workConfiguration)
+        equals("image", true) -> tagImage(resource)
```

Comment:
What about `snapshot`?

Reply:
Discussed offline with @emjburns , added a TODO for now , needs updates in `orca` and `clouddriver`